document `provider-<name>` language conformance tests (#21773)
pulumi/pulumi: cmd/pulumi-test-language/README.md

  such as custom resources, function invocations, and so on.
* *L3 tests* exercise features that require more advanced language support such
  as first-class functions -- `apply` is a good example of this.

>   * *Provider tests* exercise language code that acts as a provider, iow calls `Construct`.

Reviewer: indent looks odd here
```suggestion
* *Provider tests* exercise language code that acts as a provider, that is `Construct` and `Call` implementations.
```
